### DISCUSSION TITLE
IntuneCD on Linux or WSL

### DISCUSSION THREAD

**Discussion by aaronparker** [almenscorner/IntuneCD#121] [General]
macOS makes Python easy to use, but it's not quite so easy on Linux or WSL. I'm trying to workout how to get IntuneCD commands to be available in bash on these platforms (that is to use interactively). 

I'm attempting on Ubuntu on WSL and Linux. On Linux I've configured a venv, but on WSL I don't think that's needed. Attempting to use IntuneCD commands returns `command not found`. 

Do you know of any good guides on getting Python modules working directly in bash?

**Comment by almenscorner** [comment]
Sounds like python is missing in the PATH? I've run this without issues on WSL

**Comment by almenscorner** [comment]
For example

```bash
export PATH=”$PATH:/usr/local/bin/python
```

**Comment by aaronparker** [comment]
Excellent, that worked on WSL. Thank-you. 

Now I'm off to troubleshoot npm on WSL to get md-to-pdf working ;). I'll have to document the setup, because I'm sure to forget in the future.

**Reply by almenscorner** [reply]
Check this doc from MS in installing npm on WSL :) https://learn.microsoft.com/en-us/windows/dev-environment/javascript/nodejs-on-wsl#install-nvm-nodejs-and-npm

**Comment by aaronparker** [comment]
Homebrew to the rescue - I couldn't get md-to-pdf working on WSL or an Ubuntu VM (a Puppeteer issue), so I started again with WSL, using Homebrew to install everything and it worked.

**Comment by aaronparker** [comment]
For anyone wanting to get `IntuneCD` and `md-to-pdf` working on Ubuntu 22.04 under WSL2 on Windows 11 to backup the Intune configuration and generate a PDF document from the markdown as-built, here's the commands I used to get `Homebrew`, `IntuneCD`, `npm` and `md-to-pdf` working:

```bash
sudo apt-get update; sudo apt-get upgrade -y
sudo apt-get install -y libnss3 libnspr4 libatk1.0-0 libatk-bridge2.0-0 libcups2 libdrm2 libxkbcommon0 libxcomposite1 libxdamage1 libxfixes3 libxrandr2 libgbm1 libasound2 libpango-1.0-0 libpangocairo-1.0-0 libpangoft2-1.0-0

sudo apt-get install -y build-essential procps curl file git
/bin/bash -c "$(curl -fsSL https://raw.githubusercontent.com/Homebrew/install/HEAD/install.sh)"

python3.11 -m pip install --upgrade pip
pip3 install IntuneCD

brew instal npm jq
npm i -g npm@9.8.0
npm i -g md-to-pdf
```

You can then use this script to backup the Intune configuration and generate a PDF and HTML version of the documentation: [backup-and-doc.sh](https://github.com/aaronparker/intune-backup-template/blob/main/backup-and-doc.sh)
